api/src/pcapi/core/offers/validation.py

```diff
@@ -375,3 +379,20 @@ def check_booking_limit_datetime(
 
     if beginning and booking_limit_datetime and booking_limit_datetime > beginning:
         raise exceptions.BookingLimitDatetimeTooLate()
+
+
+def check_offer_extra_data(offer: Optional[Offer], subcategory_id: str, extra_data: dict) -> dict:
+    api_errors = ApiErrors()
+    subcategory = ALL_SUBCATEGORIES_DICT[subcategory_id]
+    mandatory_fields = list(set(OFFER_EXTRA_DATA_MANDATORY_FIELDS) & set(subcategory.conditional_fields))
```

Comment:
Pas besoin de convertir en liste. `set & set` renvoie un set, sur lequel on peut itérer directement.

```suggestion
    mandatory_fields = OFFER_EXTRA_DATA_MANDATORY_FIELDS & set(subcategory.conditional_fields))
```